include/zephyr/drivers/gpio.h

```diff
@@ -563,6 +563,8 @@ __subsystem struct gpio_driver_api {
 			       struct gpio_callback *cb,
 			       bool set);
 	uint32_t (*get_pending_int)(const struct device *dev);
+	int (*port_get_direction_raw)(const struct device *port, gpio_port_pins_t *inputs,
+					   gpio_port_pins_t *outputs);
```

Comment:
To avoid impacting the footprint when not enabled, the API struct function pointer (along with the implementation functions in all drivers) should be make conditional as shown below. This is how conditional APIs are handled in other Zephyr driver classes.

```suggestion
#ifdef CONFIG_GPIO_GET_DIRECTION
	int (*port_get_direction_raw)(const struct device *port, gpio_port_pins_t *inputs,
					   gpio_port_pins_t *outputs);
#endif /* CONFIG_GPIO_GET_DIRECTION */
```